test/sqllogictest/default_privileges.slt

```diff
@@ -3369,6 +3369,36 @@ c  materialize=U/materialize
 statement ok
 DROP CONNECTION c
 
+statement ok
+CREATE SECRET s3_api_secret_key as 'secret_key';
+
+statement ok
+CREATE CONNECTION materialize.public.a TO AWS (ACCESS KEY ID = 'access_key', SECRET ACCESS KEY = SECRET s3_api_secret_key) WITH (VALIDATE = false);
```

Comment:
@ParkMyCar Added a test that it works by default and doesn't if flag is disabled below.

Reply:
Thank you!